src/app/js/formats/vega/component/tree-map/TreeMapData.js

```diff
@@ -0,0 +1,166 @@
+export default class TreeMapData {
+    /**
+     * Raw data to be filtered and transformed
+     * @type {Array<{source:string, target: string, weight: number}>}
+     */
+    data;
+    /**
+     * Nodes and leaves ids counter
+     * @type {number}
+     */
+    idIncrement;
+    /**
+     * List of nodes and leaves ids
+     * @type {Map<string, number>}
+     */
+    ids;
+    /**
+     * Raw list of nodes and leaves ready to be filtered and transformed
+     * @type {Map<string, {parent: number?, size: number?}>}
+     */
+    rawNodesAndLeaves;
+    /**
+     * Formatted list of nodes and leaves ready to be cleanup and returned
+     * @type {Map<number, {id: number, name: string, parent: number?, size: number?}>}
+     */
+    formattedNodesAndLeaves;
+    /**
+     * Filtered nodes use to create the final transformed data
+     * @type {Set<number>}
+     */
+    filteredNodes;
+
+    constructor(data) {
+        this.data = data;
+        this.idIncrement = 0;
+        this.ids = new Map();
+        this.rawNodesAndLeaves = new Map();
+        this.formattedNodesAndLeaves = new Map();
+        this.filteredNodes = new Set();
+    }
+
+    /**
+     * Create an id to the given node if none have been created previously and return the corresponding id
+     * @param node
+     * @return {number | undefined}
+     */
+    createIdIfNotExist(node) {
+        if (!this.ids.has(node)) {
+            const id = ++this.idIncrement;
+            this.ids.set(node, id);
+            return id;
+        }
+        return this.ids.get(node);
+    }
+
+    /**
+     * Get the hierarchy of a leaf from it first parent
+     * @param node First parent of the leaf
+     * @return {string} Leaf parent list
+     */
+    createHierarchy(node) {
+        const maxRecursion = 10;
+        let recursion = 0;
+
+        const getHierarchy = (parentId) => {
+            recursion++;
+            if (recursion > maxRecursion) {
+                return [];
+            }
+
+            if (!this.formattedNodesAndLeaves.has(parentId)) {
+                return [];
+            }
+
+            const parent = this.formattedNodesAndLeaves.get(parentId);
+            if (parent.parent) {
+                return [...getHierarchy(parent.parent), parent.name];
+            }
+
+            return [parent.name];
+        };
+
+        return getHierarchy(node).join(', ');
+    }
+
+    /**
+     * Transformed routine data into transformable data
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏
```suggestion
     * Transform routine data into transformable data
```